discovery_engine_core/web-api/src/models.rs

```diff
@@ -49,6 +49,11 @@ pub(crate) enum Error {
 
 impl Reject for Error {}
 
+#[derive(Clone, Debug, Serialize)]
+struct BaseError {
```

Comment:
> I'm not sure it is worth it to have an #[serde(flatten)]-focused struct for a single fields.

in case that `BaseError` gets extended or changed, then flattening means less code changes.

>  I also don't understand why this field is added here given that request id handling was already implemented in a PR as well as the actix PR we decided to have instead. It will never be used anywhere.

the ticket was about making the implementation compliant with the specification, and the specification still has this and wasn't updated yet. i agree that it needs to be changed once the spec is updated to match the new approach, but again this is not the scope of the current ticket.

Reply:
> gets extended or changed 

But that's a premature optimization for a use case which probably will never happen.

> wasn't updated yet

It's not that request id is gone, it's just already implemented somewhere else and is handled by middle-ware instead of needing to be included into every single error type.

Reply:
removed as discussed in the daily.